src/js/pages/Index.js

```diff
@@ -41,6 +42,8 @@ var Index = React.createClass({
     MetadataStore.init();
     SidebarStore.init();
 
+    NavigationServiceUtil.registerRoutesInNavigation(this.props.routes);
```

Comment:
Not a blocker issue, but I have a feeling there is something conceptually wrong here.

I would expect a utility function to operate only on the arguments it receives and not to perform any behind-the-scenes access to instances or singletons.

For me, I would suggest either:
1. `NavigationService.registerRoutesInNavigation` or
2. `NavigationServiceUtil.registerRoutesInNavigation( NavigationService, this.props.routes );`

But again, that's not a blocker. If you agree, let's make a JIRA issue.